bgpd/bgp_route.c

```diff
@@ -3125,8 +3127,8 @@ int bgp_update(struct peer *peer, struct prefix *p, uint32_t addpath_id,
 
 			prn = bgp_node_get(bgp->rib[afi][safi],
 					   (struct prefix *)prd);
-			if (prn->info) {
-				table = (struct bgp_table *)(prn->info);
+			if (bgp_node_has_info_data(prn)) {
```

Comment:
Can be combined into one call. Get the info and check if it is NULL or not like in other places.